### ISSUE TITLE
tests/range/tcompiletime_range_checks.nim fails on aarch64

### ISSUE THREAD

**Issue by xzfc** [nim-lang/Nim#12262]
### Log
```
FAIL: tests/range/tcompiletime_range_checks.nim C
Test "tests/range/tcompiletime_range_checks.nim" in category "range"
Failure: reMsgsDiffer
Expected:
tcompiletime_range_checks.nim(36, 21) Error: 2147483648 can't be converted to int32
tcompiletime_range_checks.nim(37, 23) Error: -1 can't be converted to uint64
tcompiletime_range_checks.nim(38, 34) Error: 255 can't be converted to FullNegativeRange
tcompiletime_range_checks.nim(39, 34) Error: 18446744073709551615 can't be converted to HalfNegativeRange
tcompiletime_range_checks.nim(40, 34) Error: 300 can't be converted to FullPositiveRange
tcompiletime_range_checks.nim(41, 30) Error: 101 can't be converted to UnsignedRange
tcompiletime_range_checks.nim(42, 32) Error: -9223372036854775808 can't be converted to SemiOutOfBounds
tcompiletime_range_checks.nim(44, 22) Error: nan can't be converted to int32
tcompiletime_range_checks.nim(46, 23) Error: 1e+100 can't be converted to uint64
tcompiletime_range_checks.nim(49, 22) Error: 18446744073709551615 can't be converted to int64
tcompiletime_range_checks.nim(50, 22) Error: 18446744073709551615 can't be converted to int32
tcompiletime_range_checks.nim(51, 22) Error: 18446744073709551615 can't be converted to int16
tcompiletime_range_checks.nim(52, 21) Error: 18446744073709551615 can't be converted to int8


Gotten:
tcompiletime_range_checks.nim(36, 21) Error: 2147483648 can't be converted to int32
tcompiletime_range_checks.nim(37, 23) Error: -1 can't be converted to uint64
tcompiletime_range_checks.nim(38, 34) Error: 255 can't be converted to FullNegativeRange
tcompiletime_range_checks.nim(39, 34) Error: 18446744073709551615 can't be converted to HalfNegativeRange
tcompiletime_range_checks.nim(40, 34) Error: 300 can't be converted to FullPositiveRange
tcompiletime_range_checks.nim(41, 30) Error: 101 can't be converted to UnsignedRange
tcompiletime_range_checks.nim(42, 32) Error: -9223372036854775808 can't be converted to SemiOutOfBounds
tcompiletime_range_checks.nim(44, 22) Error: nan can't be converted to int32
tcompiletime_range_checks.nim(49, 22) Error: 18446744073709551615 can't be converted to int64
tcompiletime_range_checks.nim(50, 22) Error: 18446744073709551615 can't be converted to int32
tcompiletime_range_checks.nim(51, 22) Error: 18446744073709551615 can't be converted to int16
tcompiletime_range_checks.nim(52, 21) Error: 18446744073709551615 can't be converted to int8
```

### TL;DR

Line of interest: `let rejectJ* = (uint64(1e100))` [link](https://github.com/nim-lang/Nim/blob/6b4b329f45ce23fc7590c2c9fb3cc50715b266a1/tests/range/tcompiletime_range_checks.nim#L46)

It is expected to produce the following error, but it is not:
```
tcompiletime_range_checks.nim(46, 23) Error: 1e+100 can't be converted to uint64
```

### Additional Information

Nim version: 1.0.0 (from the official tarball)
OS: NixOS 20.03pre
CPU architecture: aarch64

**Comment by metagn** [comment]
Works now, can be closed
